### ISSUE TITLE
Support for JSR308 type annotations in nullness checker

### ISSUE THREAD

**Issue by PhilippWendler** [spotbugs/spotbugs#643]
The nullness checker of SpotBugs ignores the JSR308 `TYPE_USE` annotations like `org.checkerframework.checker.nullness.qual.Nullable`. This specific annotation is known to SpotBugs since https://github.com/spotbugs/spotbugs/commit/dd436ff56abf0273a080ad228fcf5abd8ee3a097, but it is not detected, because it seems that SpotBugs looks only for conventional annotations, not type annotations.

I am not suggesting a full analysis of type annotations that supports something like `List<@Nullable String>`, I am only suggesting that `@Nullable Object var = ...` should be supported if `@Nullable` is a type annotation instead of a regular annotation.

Test case (tested with SpotBugs 3.1.3):
```
@org.sosy_lab.common.annotations.ReturnValuesAreNonnullByDefault
public abstract class TestNullable {

  private static @org.checkerframework.checker.nullness.qual.Nullable Object getFoo() {
    return null;
  }
}
```

I do not know about the format of class files, but I suspect that type annotations are stored in a different place than regular annotations.
(As a further explanation: type annotations are also available in a different place when using Java's reflection API: for regular annotations you call e.g. `Parameter.getAnnotations()`, for type annotations you call `Parameter.getAnnotatedType().getAnnotations()`.)

Guava 25.1 has switched to using this annotation, so I suspect this will be a more common problem in the future. It has already been mentioned as a problem here: https://github.com/spotbugs/spotbugs/issues/621#issuecomment-391686616.

Note that this is not the same as #419 and #470, which suggest adding `TYPE_USE` to SpotBugs' annotations.

**Comment by slovdahl** [comment]
Is there any kind of workaround available for this problem? Is there any timeline for a SpotBugs 4.0 release?

**Comment by slovdahl** [comment]
Version 3.1.10 seems to have a fix for this, am I interpreting that right?

**Comment by olegshtch** [comment]
I got it in 4.2.0  version.

**Comment by ankilosaurus** [comment]
I am using spotbugs version 4.8.6.6 and hitting this issue with jakarta.annotation.Nullable
